docs/dev/identity-provider-tracking.md

```diff
@@ -0,0 +1,11 @@
+# Identity Provider Tracking
+Infra tracks users and groups that exist in external identity providers and maps them access. The following database diagram shows how the relation of groups and users in identity providers are mapped to users and groups within Infra.
+
+```mermaid
+erDiagram
```

Comment:
There's also an identities_group relationship that is not represented in this diagram, is that right?

Reply:
yeah sometimes the join tables are links in ERDs, but it might be useful to call it out. also I feel like you can't understand this system at all if you don't consider who owns what table. We own groups, identities, and identities_groups, and the providers own provider_users and provider_groups, and even the (implicit) providers_groups_users, which is an array on provider_users